pkg/maps/configmap/configmap.go

```diff
@@ -135,7 +135,7 @@ type endpoint interface {
 
 // EndpointConfigMap is a map type for interfacing with endpoint BPF config.
 type EndpointConfigMap struct {
-	Map  *bpf.Map
+	*bpf.Map
```

Comment:
Leave the `Map` because this way you will be able to know if `Map` is nil or not and don't get a nil pointer when doing the close on line 1451 from the other file.

Reply:
>What do you mean by "Leave the Map"?

Leave the `Map` attribute as it was before this change.

>The Map field inside EndpointConfigMap should always be non-nil (it's initialised in OpenMapWithName(), and the reference in the Endpoint refers to a *EndpointConfigMap, which may or may not be nil but the check in the other file should be fine in that regard.

That is true but there's also a problem if we later on add the "Close()" method to `EndpointConfigMap` which will overwrite all usages of `bpf.Map.Close()` without us knowing

Reply:
*shrug That's how Golang works. It's more boilerplate that's more explicit if we define a `Close()`, which helps non-Go programmers to contribute I guess. So long as the core implementation is shared I don't mind.

Reply:
I'm fine with this as-is as well.